frontend/src/metabase/querying/filters/components/FilterPicker/types.ts

```diff
@@ -1,22 +1,21 @@
 import type * as Lib from "metabase-lib";
 
-export interface FilterPickerWidgetProps {
+export type FilterPickerWidgetProps = {
   query: Lib.Query;
   stageIndex: number;
   column: Lib.ColumnMetadata;
   filter?: Lib.FilterClause;
   isNew: boolean;
-  onChange: (filter: Lib.ExpressionClause) => void;
+  withAddButton: boolean;
+  onChange: (filter: Lib.ExpressionClause, opts: FilterChangeOpts) => void;
   onBack?: () => void;
-}
+};
 
-export interface PickerOperatorOption<Operator> {
-  operator: Operator;
+export type FilterChangeSource = "default" | "add-button";
 
-  // An operator's longDisplayName is going to be used by default,
-  // but widgets can overwrite it with a custom name.
-  name?: string;
-}
+export type FilterChangeOpts = {
+  source: FilterChangeSource;
+};
```

Comment:
I'm not sure if this abstraction is worth it. The children components sort of already know (through labels and tooltips) that clicking "Add another filter" won't run the query and clicking "Apply filter" will.

`MultiStageFilterPicker` would be simpler if it didn't have to translate `FilterChangeSource` to `QueryChangeOpts`.

Also it's unclear what `"default"` means, and it's not immediately apparent which button is the `"add-button"` (they both are "add" buttons, one just happens to have a plus icon - "add-more-button" would be more accurate).

```suggestion
export type FilterChangeOpts = QueryChangeOpts;
```

Reply:
I removed this abstraction and passed `run: boolean` directly